mono/utils/memfuncs.h

```diff
@@ -23,5 +23,10 @@ void mono_gc_memmove_atomic (void *dest, const void *src, size_t size);
 void mono_gc_memmove_aligned (void *dest, const void *src, size_t size);
 guint64 mono_determine_physical_ram_size (void);
 guint64 mono_determine_physical_ram_available_size (void);
+#if defined (__FreeBSD__) || defined (__linux__) || defined (__APPLE__) 
```

Comment:
I'd prefer if we extract this into a `HAVE_CGROUP_SUPPORT` define which we e.g. define in autoconf on these platforms rather than hardcoding them here.